Release/0.9.0 - all changes

shakeelmohamed opened the pull request:
Please review @glennblock!
### Do not merge until https://github.com/splunk/splunk-javascript-logging/pull/2 is merged

Commit:
Add retry on error logic, via exponential backoff.

Specify the max number of retries when configuring
a logger. The maximum backoff (time between retries)
is capped at 2 minutes.

Commit:
Add retry example

Commit:
Update examples

Commit:
Ensure manual flushing works correctly

Commit:
Add support for batch intervals

Commit:
Update tests with new expected behavior for batch settings

Commit:
Add support for maxBatchSize config setting

Commit:
Add configuration setting for batch size

Commit:
remove middleware and all references to it

We have collectively decided that this feature
does not provide enough value at this time.

Commit:
Align implementation & tests with splunk-logging

Commit:
Update old examples

Commit:
Rename batching example to manual_batching

Commit:
Update test + docstrings

Commit:
Add all_batching and custom_format examples

Commit:
Update version number to 0.9.0; changelog

glennblock commented on CHANGELOG.md:
```diff
@@ -1,5 +1,29 @@
 # Splunk HTTP Event Collector Stream for Bunyan (beta)
```
We should drop using the "beta" term here. .8 was the beta milestone.

glennblock commented on examples/manual_batching.js:
```diff
@@ -32,16 +29,12 @@ var bunyan = require("bunyan");
-    port: 8088,
-    level: "info",
-    autoFlush: false
+    token: "56A70C3F-80C1-4879-B11F-BF043B52753F",//"your-token-here",
```
shouldn't value here just be "your-token-here"?

shakeelmohamed replied:
good catch!

glennblock commented:
Overall LGTM, though a few comments

Commit:
Remove real token from example

This token is unique to a Splunk
instance, this commit is made
for consistency with other examples.

Commit:
Remove "beta" phrasing

Commit:
Update wording

shakeelmohamed committed:
Merge pull request #4 from splunk/fix/wording

Update wording

Commit:
Update retry example to retry 10 times
